src/opensles/AudioInputStreamOpenSLES.cpp

```diff
@@ -253,9 +253,10 @@ Result AudioInputStreamOpenSLES::requestStart() {
             break;
     }
 
-    if (mStreamCallback != nullptr) { // Was a callback requested?
-        setDataCallbackEnabled(true);
-    }
+    // We use a callback if the user requests one
+    // OR if we have an internal callback to fill the blocking IO buffer.
+    setDataCallbackEnabled(true);
```

Comment:
I am not familiar with oboe code base however In my opinion, if we always set dataCallbackEnabled as true on starting stream then we better get rid of it. The state(mDataCallbackEnabled) is not going to change mid stream and will never be false. Therefore we should remove it to simplify.

Reply:
It is not always set true. It is set conditionally by AAudio. This was added to workaround some old bugs in the framework. And it does change mid stream. So we still need it.